test/common/router/router_test.cc

```diff
@@ -1141,6 +1141,49 @@ TEST_F(RouterTest, ResetDuringEncodeHeaders) {
   EXPECT_TRUE(verifyHostUpstreamStats(0, 1));
 }
 
+TEST_F(RouterTest, UpstreamTimeoutStats) {
+  TestScopedRuntime scoped_runtime;
+  Runtime::LoaderSingleton::getExisting()->mergeValues(
+      {{"envoy.reloadable_features.do_not_await_headers_on_upstream_timeout_to_emit_stats",
+        "false"}});
```

Comment:
I think you want to have a test that wouldn't have raised a timeout previously but does with the new code, and then show that disabling the guard does not raise the timeout?

Reply:
Sorry, @mattklein123. For some reason not all the changes I made got synced. I just pushed the new commit with all the changes.